tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/CodeownersHelper.cs

```diff
@@ -0,0 +1,587 @@
+                {
+                    return entry;
+                }
+            }
+            return null;
+        }
+
+        public (CodeownersEntry entry, bool existed) CreateOrUpdateCodeownersEntry(
+            CodeownersEntry? existingEntry,
+            string path,
+            string serviceLabel,
+            List<string> serviceOwners,
+            List<string> sourceOwners,
+            bool isAdding,
+            bool isMgmtPlane)
+        {
+            if (existingEntry != null)
+            {
+                // Update existing entry with new codeowners
+                if (isAdding)
+                {
+                    existingEntry.ServiceOwners = AddUniqueOwners(existingEntry.ServiceOwners, serviceOwners);
+                    existingEntry.SourceOwners = AddUniqueOwners(existingEntry.SourceOwners, sourceOwners);
+                }
+                else
+                {
+                    existingEntry.ServiceOwners = RemoveOwners(existingEntry.ServiceOwners, serviceOwners);
+                    existingEntry.SourceOwners = RemoveOwners(existingEntry.SourceOwners, sourceOwners);
+                }
+
+                return (existingEntry, true);
+            }
+            else
+            {
+                // Create new entry
+                if (string.IsNullOrEmpty(serviceLabel) || string.IsNullOrEmpty(path))
+                {
+                    throw new Exception($"When creating a new entry, both a Service Label and Path are required. Provided: serviceLabel = '{serviceLabel}', path = '{path}'");
+                }
+
+                List<string> serviceLabels = new List<string>() { serviceLabel };
+
+                if (isMgmtPlane)
+                {
+                    serviceLabels.Add("Mgmt");
+                }
+
+                var newEntry = new CodeownersEntry()
+                {
+                    PathExpression = path ?? string.Empty,
+                    ServiceLabels = serviceLabels ?? new List<string>(),
+                    PRLabels = new List<string>() { serviceLabel} ?? new List<string>(),
+                    ServiceOwners = serviceOwners ?? new List<string>(),
+                    SourceOwners = sourceOwners ?? new List<string>(),
+                    AzureSdkOwners = new List<string>()
+                };
+                return (newEntry, false);
+            }
+        }
+
+        private string NormalizeLabel(string input)
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                return string.Empty;
+            }
+            return input.Replace(" ", "").Replace("%", "").Trim();
+        }
+
+        public string NormalizePath(string path)
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                return string.Empty;
+            }
+
+            path = path.Trim('/').Trim();
```

Comment:
Maybe `path.Trim().Trim('/')` to handle cases like " /sdk/service/library "

The way it's written today would normalize something like 

" /sdk/service/library " to "/sdk/service/library" 


Maybe the normalized input should be validated to be of the form "sdk/service/library" and an exception is thrown if the input can't be normalized to meet the intended format.